### PR TITLE
Setting up E-2-E Tests to run locally

### PR DESCRIPTION
**Pull request by kriscfoster**
Setting up E-2-E Tests for Timeline-Bot to run locally

- This is just part of a spike and is by no means a finished solution, just a conversation starter.
- Current workflow for running tests ->

1. From local Refocus use the command: ```npm run start-bot-integration-test``` (This cleans the refocus db in preparation for the tests, & starts refocus) - Waiting on https://github.com/salesforce/refocus/pull/1155

2. From the bot use the command: ```npm run test:end-to-end```
- This retrieves an API token from refocus
- Appends it to .env file
- Starts/installs the bot using API token
- Creates a roomtype & room with the bot
- Runs the E-2-E tests

(See gif below for this in action)

e-2-e-clean

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
     "babel-preset-es2015": "^6.24.0",
     "babel-preset-react": "^6.24.1",
     "babel-preset-stage-0": "^6.22.0",
+<<<<<<< HEAD
```
Comment (BrandonMurnane) [inline package.json]: rebase?

File: package.json [hunk 2]
```diff
@@ -64,17 +65,33 @@
     "eslint-plugin-import": "^2.11.0",
     "eslint-plugin-jsx-a11y": "^6.0.3",
     "eslint-plugin-react": "^7.7.0",
+=======
```
Comment (BrandonMurnane) [inline package.json]: as above

File: package.json [hunk 3]
```diff
@@ -64,17 +65,33 @@
...
+    "css-loader": "^2.1.1",
+    "eslint": "^4.19.0",
+    "eslint-config-airbnb": "^16.1.0",
+    "eslint-plugin-import": "^2.17.2",
+    "eslint-plugin-jsx-a11y": "^6.2.1",
+    "eslint-plugin-react": "^7.12.4",
+>>>>>>> f6e91ee6be6f1ba03a62c9154349447ba283a015
```
Comment (BrandonMurnane) [inline package.json]: as above

File: README.md [hunk 4]
```diff
@@ -5,13 +5,16 @@ A Refocus bot that show all events in a room.
...
 
 ### Env Variables
-
 Note: If you want to test this locally you will need some environment variables:
 * ```API_TOKEN``` - Used for Requests to Refocus. Created in refocus/tokens/new.
 * ```SOCKET_TOKEN``` (Returned Upon Installation) - Used for Socket Connection.
 * ```NODE_ENV (defaults to 'dev')``` - Used to determine which instance of Refocus to install the bot.
 * ```REFOCUS_URL``` (OPTIONAL) - Used to specify which refocus instance to point to.
 
+### Running E-2-E Tests With Refocus Locally
```
Comment (BrandonMurnane) [inline README.md]: can we get a little explanation into what we are using for techs

File: index.js [hunk 5]
```diff
@@ -7,24 +7,10 @@
...
 
-// Installs / Updates the Bot
 bdk.installOrUpdateBot(packageJSON);
-
-app.use(express.static('web/dist'));
-app.get('/*', (req, res) => {
-  res.sendFile(path.join(__dirname, '/web/dist/index.html'));
-});
-
-http.Server(app).listen(PORT, () => {
```
Comment (BrandonMurnane) [inline index.js]: where are you starting server now ?

Comment (kriscfoster) [reply index.js]: Server doesn't need to start for UI bots

**Comment by kriscfoster** [comment]
@BrandonMurnane Let's have a chat after standup about these

File: README.md (hunk 4 above)
Comment (kriscfoster) [reply README.md]: @BrandonMurnane is this what you meant?

Comment (BrandonMurnane) [reply README.md]: yeah looks good

**Comment by kriscfoster** [comment]
accidently merged, I've reverted and am creating a new PR

**Pull request merged**